src/sql-parser/tests/testdata/raise

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 sqlparser-rs contributors. All rights reserved.
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# This file is derived from the sqlparser-rs project, available at
+# https://github.com/andygrove/sqlparser-rs. It was incorporated
+# directly into Materialize on December 21, 2019.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the LICENSE file at the
+# root of this repository, or online at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+parse-statement
+RAISE NOTICE
+----
+RAISE NOTICE
+=>
+Raise(RaiseStatement { severity: Notice, code: None, detail: None, hint: None })
+
+parse-statement
+RAISE NOTICE CODE '01234' DETAIL 'detail' HINT 'hint'
+----
+RAISE NOTICE CODE '01234' DETAIL 'detail' HINT 'hint'
+=>
+Raise(RaiseStatement { severity: Notice, code: Some("01234"), detail: Some("detail"), hint: Some("hint") })
```

Comment:
If I'm understanding the PostgreSQL docs correctly, the way to format this in PostgreSQL would be:
`RAISE NOTICE USING ERRCODE = '01234', DETAIL = 'detail', HINT = 'hint'`.

https://www.postgresql.org/docs/15/plpgsql-errors-and-messages.html

Is there a specific reason we're diverging here?

Reply:
Haha I didn't realize this was a real postgres command; thought we made it up. Will fix.

Reply:
It's actually a plpgsql command, so you can't just issue a standalone `RAISE ....` via `psql`. You have to do something like this:
```
postgres=# DO $$
BEGIN
RAISE NOTICE USING ERRCODE = '01234', DETAIL = 'detail', HINT = 'hint';
END;
$$ LANGUAGE plpgsql;
NOTICE:  01234
DETAIL:  detail
HINT:  hint
DO
```

Though we allow it as a normal SQL command, so it doesn't match perfectly. Still we should probably stick with the same syntax.